### DISCUSSION TITLE
Html from Evernote : example for testdata helpful?

### DISCUSSION THREAD

**Discussion by Shampra** [anyproto/contributors#60] [Anytype apps]
Would it be useful to share an HTML page exported from Evernote?
The html test page (core/block/import/html/testdata/test.html) is very... basic 😅.
But I don't know if it's useful, or even if the planned Evernote import (or used on the team side according to the last Town hall) is via html?

**Comment by fuksman** [comment]
Hi @Shampra, we plan to introduce JSON import soon. Once this feature is available, contributors will be able to build imports from different services using JSON import. 
However, before we release this feature, it is already possible for anyone in the contributors' community to develop a utility that can take data exported from Evernote and convert it to protofiles based on the `any-block` format. These protofiles can then be imported into your space using the existing function. 
If you're interested, we can launch an experimental project for contributors. You can provide data examples, and I can make an announcement for contributors.

**Reply by Shampra** [reply]
Here, the idea was to improve the existing HTML import, in particular image/files integrations.
I've looked at the code, I'd go for it, but having never done Go dev, it's going to take me longer to do something (and even longer to do it right ^^).
Without knowing how to help, I just test each version with my template page. And since there's an html page template in the core repository, that explains this post ;-).

I have several questions:
- Would the imports you mention replace the existing HTML import, or does this one also need to be improved?
- At the Town Hall, @charlotte-chiang  answered a question about the Evernote import by saying that a more reliable import is used internally, as I understand it. Is this the case and not available on the repository? Are there any plans to share it?
That's my fear: contributing to development on things already done or in progress (so useless help).

To respond to the proposal...
Thank you for it @fuksman.
I'm motivated to do something other than translation, but I don't know what I could do. So even though I'm eager to see AT evolve, there's no hurry, it'll come when you get to it. And when you suggest something, I'll obviously take a look. I don't know yet if and what (and when 😁) I could do (I'm already going to look at what these famous protofiles are 😅).

**Reply by fuksman** [reply]
1. New imports will not improve HTML import, but they will create a more sophisticated importing options. I don't think that HTML import, even if improved, will be sufficient for importing complex structures. JSON/PB will work better here.
2. Hmm, I think it was about JSON or PB imports. They are more reliable.

Speaking about contribution: As I see in the documentation, Evernote should provide exports not only in HTML format but also in ENEX format. Could you please share an example of such an export, or in some other formats if available? No rush here.

**Reply by Shampra** [reply]
Can't wait!

I've never looked at what an Enex is, now's my chance.
Here is a package with Enex and HTML export of the same page (with almost all the items available on Evernote... French version, it allows you to see the problems of importing accents in the same time 😁)

[Demo Export Evernote for Anytype.zip](https://github.com/anyproto/contributors/files/12312408/Demo.Export.Evernote.for.Anytype.zip)

**Comment by SilverBut** [comment]
For @fuksman, thanks for the JSON/PB import feature! Some related question about content importing.

I'm writing something to import from Confluence to Anytype, and just finished reading Anytype's code. The tool would be written in Typescript as a separate application. For now, I want to convert the Confluence XML export format into a zip of protobuf files. But I'm uncertain about it. Should I export into protobuf, or I should call the gRPC endpoint opened by `anytypeHelper`? Any suggestions?

**Reply by fuksman** [reply]
@SilverBut, cool! Our very first import, created by contributors, is on its way! :)

I think it will be easier for you to implement the conversion from Confluence XML to Anytype PB (or JSON) and ask users to import the result into their spaces manually. I wouldn't recommend triggering any endpoints as we will deliver a URL schema for imports later, and it will be easier to use.
